### PR TITLE
fix podhttpchaos: check observedGeneration to avoid duplicated applying

### PR DESCRIPTION
**Pull request by Hexilee**
Signed-off-by: xixi <i@hexilee.me>

### What problem does this PR solve?

Now, the controller of PodHttpChaos applies rules in each time reconciling. We should check the observedGeneration to avoid duplicated applying.

### What's changed and how it works?

Add checking block:

```go
	if obj.ObjectMeta.Generation <= obj.Status.ObservedGeneration && obj.Status.FailedMessage == "" {
		r.Log.Info("the target pod has been up to date", "pod", obj.Namespace+"/"+obj.Name)
		return ctrl.Result{}, nil
	}
```

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [x] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: controllers/podhttpchaos/controller.go [hunk 1]
```diff
@@ -58,6 +58,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	if obj.ObjectMeta.Generation <= obj.Status.ObservedGeneration && obj.Status.FailedMessage == "" {
```
Comment (YangKeao) [inline controllers/podhttpchaos/controller.go:61]: Maybe we could write `PodXXXChaosDelegate` (with a better name), as the controllers of `podiochaos`, `podhttpchaos`, and `podnetworkchaos` are nearly the same in the future.

This PR is fine to merge.

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by fewdan — APPROVED** [review]
LGTM

File: controllers/podhttpchaos/controller.go (hunk 1 above)
Comment (Hexilee) [reply controllers/podhttpchaos/controller.go:61]: Looks good

**Pull request merged**